### PR TITLE
Create the CheckoutPage

### PR DESCRIPTION
**Pull request by CharlotteW923**
## Description

Created the Checkout Page

## Changes

- Added the checkout page with all the necessary forms and pay button
- Made Select a state a dropdown menu
- When you click on Pay with stripe it just console logs all the data you inputted

## Screenshots/demo

image

Clicking the pay button without entering required info:
image
image
Console log after clicking pay:
image

## Related issues

What issue(s) does this PR fix?
#87

### CODE REVIEW

**Review by sebastian-nunez — APPROVED** [review]
Awesome work! Just a couple minor things. Once you merge I'll work on hooking the flow from selecting an item and putting it in the cart to completing the checkout form to paying with stripe and being redirected to the confirmation page :)

File: web-client/src/pages/CheckoutPage.tsx [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+                                    </FormItem>
+                                )}
+                            />
+                        </div>
+
+                        <p className="text-gray-500 mt-8 mb-8">
+                            Shipping information and order updates will be sent to your email address.
+                        </p>
+
+                        <Button type="submit" className="w-full bg-black text-white">Pay with Stripe</Button>
```
Comment (sebastian-nunez) [inline web-client/src/pages/CheckoutPage.tsx:204]: You can remove the custom styles and just leave the default button :)

Comment (sebastian-nunez) [inline web-client/src/pages/CheckoutPage.tsx:204]: You can remove the custom styles and just leave the default button :)

**Review by sebastian-nunez — APPROVED** [review]

File: web-client/src/pages/CheckoutPage.tsx [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+
+const checkoutSchema = z.object({
+    firstname: z.string().min(1, "First name is required").max(32, "First name must be 32 characters or less"),
+    lastname: z.string().min(1, "Last name is required").max(32, "Last name must be 32 characters or less"),
+    email: z.string().email("Invalid email address"),
+    phone: z.string().min(10, "Phone number must be at least 10 digits").max(16, "Phone number must be 16 digits or less").optional(),
+    street: z.string().min(1, "Street address is required").max(100, "Street address must be 100 characters or less"),
+    apt_unit: z.string().max(16, "Apartment/Unit must be 16 characters or less").optional(),
```
[the remark this answers is not in the document]
Comment (CharlotteW923) [reply web-client/src/pages/CheckoutPage.tsx:28]: Fixed! Also changed it on lines 44 and 142

**Comment by sebastian-nunez** [comment]
I'm going to merge this PR now, so I can do some fixes for the overall project. You can just commit the other changes later directly into develop

File: web-client/src/pages/CheckoutPage.tsx (hunk 1 above)
Comment (CharlotteW923) [reply web-client/src/pages/CheckoutPage.tsx:204]: Fixed!

Comment (CharlotteW923) [reply web-client/src/pages/CheckoutPage.tsx:204]: Fixed!

**Pull request merged**